core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -547,31 +522,11 @@ private static String partitionsToLogString(Collection<TopicIdPartition> partiti
         return ShareSession.partitionsToLogString(partitions, log.isTraceEnabled());
     }
 
-    /**
-     * Recursive function to process all the fetch requests present inside the fetch queue
-     */
     // Visible for testing.
-    void maybeProcessFetchQueue() {
-        if (!acquireProcessFetchQueueLock()) {
-            // The queue is already being processed hence avoid re-triggering.
-            return;
-        }
-
-        ShareFetchData shareFetchData = fetchQueue.poll();
-        if (shareFetchData == null) {
-            // No more requests to process, so release the lock. Though we should not reach here as the lock
-            // is acquired only when there are requests in the queue. But still, it's safe to release the lock.
-            releaseProcessFetchQueueLock();
-            return;
-        }
-
+    void processShareFetch(ShareFetchData shareFetchData) {
```

Comment:
While you are at this class, in the following code, `return` seems unnecessary. 

```
                sharePartition.maybeInitialize().whenComplete((result, throwable) -> {
                    if (throwable != null) {
                        maybeCompleteInitializationWithException(sharePartitionKey, shareFetchData.future(), throwable);
                        return;
                    }
```

Reply:
Done. Though I am reafactoring the initilization handling in Share Partition Manager as we need to address the concern when Share Partition is loaded late and request is added to purgatory, etc. The PR will be out soon.